### DISCUSSION TITLE
New to training AI models? We've got you covered! 🧠

### DISCUSSION THREAD

**Discussion by NLadi** [ultralytics/ultralytics#22424] [Show and tell]
In her latest blog, Abirami Vina breaks down the basics of training an AI model. Whether you're just starting out or looking for a quick refresher, this beginner-friendly guide walks you through the core steps and tools.

Learn more ➡️ https://www.ultralytics.com/blog/a-quick-guide-for-beginners-on-how-to-train-an-ai-model

<a href="https://www.ultralytics.com/blog/a-quick-guide-for-beginners-on-how-to-train-an-ai-model" target="_blank">
  <img width="1364" height="790" alt="Guide_Thumbnail" src="https://github.com/user-attachments/assets/f8f1987e-aa62-4d96-80b9-7d549bf429cc" />
</a>

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NLadi, thanks for starting this discussion and for highlighting beginner-friendly resources 🚀 This is an automated response to help you get oriented quickly—an Ultralytics engineer will also assist you here soon.

If you’re new to training models, we recommend a visit to the [Docs](https://docs.ultralytics.com/) where you’ll find step-by-step [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and answers to many frequently asked questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please include as much detail as possible, such as dataset samples and logs, and review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best:
- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions on [Discourse](https://community.ultralytics.com/)
- Community threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

Please upgrade to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to ensure you’re on the most recent version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Thanks for sharing, @NLadi—here’s a 60‑second hands-on start with Ultralytics YOLO11:

```bash
# CLI: quick train on the built-in COCO8 sample and then run inference
yolo train model=yolo11n.pt data=coco8.yaml epochs=10
yolo predict model=runs/detect/train/weights/best.pt source=path/to/images
```

```python
# Python API
from ultralytics import YOLO
model = YOLO("yolo11n.pt")
model.train(data="coco8.yaml", epochs=10)
model.predict(source="path/to/images")
```

More details are in the [Train mode guide](https://docs.ultralytics.com/modes/train/) and our [Ultralytics Guides](https://docs.ultralytics.com/guides/). If you hit an error, run `yolo checks` and paste the text output here so we can help.
